cfme/tests/ssui/test_ssui_myservice.py

```diff
@@ -295,7 +298,50 @@ def test_retire_owned_service():
             3.
             4. Service should retire
     """
-    pass
+    user, _ = user_self_service_role
+
+    # login with user having self service role
+    with user:
+        with appliance.context.use(context):
+            appliance.server.login(user)
+
+            # order service from catalog item
+            serv_cat = ServiceCatalogs(
+                appliance,
+                catalog=generic_catalog_item.catalog,
+                name=generic_catalog_item.name,
+            )
+
+            if context == ViaSSUI:
+                serv_cat.add_to_shopping_cart()
+
+            provision_request = serv_cat.order()
+            provision_request.wait_for_request()
+            service = MyService(appliance, generic_catalog_item.dialog.label)
+
+            @request.addfinalizer
+            def _clear_request_service():
+                if provision_request.exists():
+                    provision_request.remove_request(method="rest")
+                if service.exists:
+                    service.delete()
+
+            assert service.exists
+
+            # Retire service
+            retire_request = service.retire()
+
+            @request.addfinalizer
+            def _clear_retire_request():
+                if retire_request.exists():
```

Comment:
I think it this test could be made slightly more powerful by changing the `if retire_request.exists():` to `assert retire_request.exists`